src/sync/sync_permission.cpp

```diff
@@ -61,6 +61,38 @@ Permission::AccessLevel extract_access_level(Object& permission, CppContext& con
 
 }
 
+// MARK: - Permission
+
+std::string Permission::description_for_access_level(AccessLevel level)
+{
+    switch (level) {
+        case AccessLevel::None: return "none";
+        case AccessLevel::Read: return "read";
+        case AccessLevel::Write: return "write";
+        case AccessLevel::Admin: return "admin";
+    }
+    REALM_UNREACHABLE();
+}
+
+bool Permission::paths_are_equivalent(std::string path_1, std::string path_2,
+                                      std::string user_id_1, std::string user_id_2)
+{
+    if (path_1 == path_2) {
+        // If both paths have `/~/`, the user IDs must match.
+        return (path_1.find("~") == npos) ? true : (user_id_1 == user_id_2);
```

Comment:
I think this comparison should be against `std::string::npos` rather than `realm::npos` as it currently.